packages/next/src/server/app-render.tsx

```diff
@@ -2026,7 +2031,19 @@ export async function renderToHTMLOrFlight(
           ...validateRootLayout,
         })
 
-        return result
+        const escapeQuote = (str: string) => str.replace(/"/g, '&quot;')
+        const injected = nextInjectToStreamBuffer
+          .map((obj) => {
+            const div = `<div id="${escapeQuote(
+              obj.id
+            )}" data-value="${escapeQuote(
+              obj.value
+            )}">Michal is the greatest</div>`
+            console.log(`Injecting into stream: ${div}`)
+            return div
+          })
+          .join('')
+        return result.pipeThrough(createSuffixStream(injected))
```

Comment:
Moving this out will make this quite painless!
```suggestion
        const injected = nextInjectToStreamBuffer.join('')
        return result.pipeThrough(createSuffixStream(injected))
```